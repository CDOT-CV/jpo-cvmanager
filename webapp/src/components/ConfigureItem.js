import React, { Component } from 'react';

import './css/ConfigureItem.css';

class ConfigureItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: false,
    }
  }

  toggleSelect = () => {
    let select = !(this.state.selected)
    this.props.updateRsu(this.props.index, select)
    this.setState({'selected': select})
  }

  render() {
    return (
      <div id={this.props.indexList.includes(this.props.index) ? "selectedconfigitemdiv" : "configitemdiv"} 
            onClick={() => this.toggleSelect()}>
        <p id="configitemtext">
          <strong>{this.props.ip}</strong>
        </p>
      </div>
    )
  }
}

export default ConfigureItem;