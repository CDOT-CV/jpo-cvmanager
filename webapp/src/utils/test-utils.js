/**
 * This function searches for the every react-aria SSR ids in a given HTMLElement node and replace every attribute values with a static id
 *
 * This can be usefull when you're trying to generate a snapshot of components using react-aria under the hood
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceReactAriaIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: "class",
      updateFunc: (val) =>
        val.replace(
          /css-[0-9a-z]{6}-MuiTableCell-root-MuiTablePagination-root/g,
          "css-mocked-MuiTableCell-root-MuiTablePagination-root"
        ),
    },
  ];

  container.querySelectorAll("td").forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)));
      }
    });
  });
  container.querySelectorAll("input[aria-invalid]").forEach((item) => {
    delete item["aria-invalid"];
  });
  return container;
}

export { replaceChaoticIds };
