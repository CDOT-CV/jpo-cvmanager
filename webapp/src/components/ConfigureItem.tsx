import React, { useState } from 'react'

import './css/ConfigureItem.css'

interface ConfigureItemProps {
  ip: string
  index: number
  indexList: number[]
  updateRsu: (index: number, selected: boolean) => void
}

const ConfigureItem = (props: ConfigureItemProps) => {
  const [selected, setSelected] = useState(false)

  const toggleSelect = () => {
    let localSelect = !selected
    props.updateRsu(props.index, localSelect)
    setSelected(localSelect)
  }

  return (
    <div
      id={props.indexList.includes(props.index) ? 'selectedconfigitemdiv' : 'configitemdiv'}
      onClick={() => toggleSelect()}
    >
      <p id="configitemtext">
        <strong>{props.ip}</strong>
      </p>
    </div>
  )
}

export default ConfigureItem
