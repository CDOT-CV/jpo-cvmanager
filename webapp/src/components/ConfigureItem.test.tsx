import React from 'react'
import { render, screen, fireEvent, queryByAttribute } from '@testing-library/react'
import ConfigureItem from './ConfigureItem'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const updateRsu = jest.fn()
  const { container } = render(<ConfigureItem indexList={[0]} updateRsu={updateRsu} index={0} ip={''} />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()

  const getById = queryByAttribute.bind(null, 'id')

  fireEvent.click(getById(container, 'selectedconfigitemdiv'))
  expect(updateRsu).toHaveBeenCalledTimes(1)
  expect(updateRsu).toHaveBeenCalledWith(0, true)

  fireEvent.click(getById(container, 'selectedconfigitemdiv'))
  expect(updateRsu).toHaveBeenCalledTimes(2)
  expect(updateRsu).toHaveBeenCalledWith(0, false)
})
