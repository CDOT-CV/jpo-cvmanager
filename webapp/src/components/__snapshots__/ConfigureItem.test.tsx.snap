// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="selectedconfigitemdiv"
  >
    <p
      id="configitemtext"
    >
      <strong />
    </p>
  </div>
</div>
`;
